packages/zendesk-adapter/src/filters/guide_default_language_settings.ts

```diff
@@ -0,0 +1,109 @@
+/*
+*                      Copyright 2022 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import {
+  Change,
+  getChangeData,
+  InstanceElement, isAdditionChange, isAdditionOrModificationChange,
+  isInstanceChange,
+  isInstanceElement, isModificationChange,
+} from '@salto-io/adapter-api'
+import { client as clientUtils } from '@salto-io/adapter-components'
+import _ from 'lodash'
+import { FilterCreator } from '../filter'
+import { FETCH_CONFIG } from '../config'
+import { BRAND_TYPE_NAME, GUIDE_LANGUAGE_SETTINGS_TYPE_NAME } from '../constants'
+import { getZendeskError } from '../errors'
+
+export const DEFAULT_LOCALE_API = '/hc/api/internal/default_locale'
+
+/**
+ On fetch - Add 'default' field for guide_language_settings from an url request
+ On deploy - ignore the field
+ */
+const filterCreator: FilterCreator = ({ config, client, brandIdToClient = {} }) => ({
+  onFetch: async elements => {
+    if (!config[FETCH_CONFIG].enableGuide) {
+      return
+    }
+    const instances = elements.filter(isInstanceElement)
+
+    const brands = instances
+      .filter(e => e.elemID.typeName === BRAND_TYPE_NAME)
+      .filter(b => b.value.has_help_center === true)
+
+    const brandToDefaultTranslation: Record<number, clientUtils.ResponseValue | clientUtils.ResponseValue[]> = {}
+
+    // Request the default locale for each brand
+    const defaultLocaleRequestPromises = brands.map(async brand => {
+      const brandId = brand.value.id
+      const res = await brandIdToClient[brandId].getSinglePage({ url: DEFAULT_LOCALE_API })
+      brandToDefaultTranslation[brandId] = res.data
+    })
+
+    // Do all requests parallel to save time
+    await Promise.all(defaultLocaleRequestPromises)
```

Comment:
1. please use functional-style implementations - specifically here this can also save the need to define the initialized type
2. we shouldn't have anything relying on `clientUtils.ResponseValue` here, it should remain at the client level - we should validate the type of the response and extract just what we need
3. it's nice that we can parallelize, but in this case since it's one request per brand I think we don't need to optimize it just yet

so this can be something like
```suggestion
    const brandToDefaultTranslation = Object.fromEntries(await awu(brands).map(brand => [brand.value.id, await getDefaultLocale(brand.value.id)]]))
```

where `getDefaultLocale` is guaranteed to return a string (and if not then we handle the edge case)